core/src/main/java/org/elasticsearch/index/analysis/CustomAnalyzerProvider.java

```diff
@@ -88,6 +94,34 @@ public void build(final Map<String, TokenizerFactory> tokenizers, final Map<Stri
         );
     }
 
+    public static TokenFilterFactory checkAndApplySynonymFilter(TokenFilterFactory tokenFilter, String tokenizerName, TokenizerFactory tokenizer,
+                                                                List<TokenFilterFactory> tokenFilterList,
+                                                                List<CharFilterFactory> charFiltersList,
+                                                                int positionIncrementGap, int offsetGap, Environment env) {
```

Comment:
I'm not sure what you say. The value from https://github.com/elastic/elasticsearch/pull/8049/files/a297d89134c18fad0daf69b2c48356d4c09a6764#diff-1ea188dabb8bc41bc2812d1564604899R84 is not the default value. Should not use the value from analyze_settings always?